aspnetcore/performance/caching/middleware.md

```diff
@@ -1,15 +1,104 @@
+--- | ---
+UseCaseSensitivePaths | <p>Determines if responses will be cached on case-sensitive paths.</p><p>The default value is `false`.</p>
+MaximumBodySize | <p>The largest cacheable size for the response body in bytes.</p>The default value is `64 * 1024 * 1024` [64 MB (67,108,864 bytes)].</p>
+
+The following example configures these options so that the middleware will independently cache responses on case-sensitive paths and on body response size. Configured as shown below, the middleware would independently cache the responses for `/page1` and `/PaGe1`. The middleware would also only cache responses that have a body size less than 1 MB (1,024 bytes).
+```c#
+services.AddResponseCaching(options =>
+{
+    options.UseCaseSensitivePaths = true;
+    options.MaximumBodySize = 1024;
+});
+```
+
+## ResponseCache attribute
+The `ResponseCache` attribute specifies the parameters necessary for setting appropriate headers for response caching. The only parameter of the `ResponseCache` attribute that strictly requires the middleware is `VaryByQueryKeys`, which does not correspond to an actual HTTP header. See the [ResponseCache Attribute document](response.md#responsecache-attribute) and the [ResponseCacheAttribute Class API Reference](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.aspnetcore.mvc.responsecacheattribute) for usage and a description of the parameters.
+
+## Caching conditions
+* The request must result in a 200 (OK) response from the server.
+* The request method must be GET or HEAD.
+* Terminal middlewares, such as Static File Middleware, must not process the response prior to the Response Caching Middleware.
+* The Authorization header must not be present.
+* `Cache-Control` header parameters must be valid and the response must be marked `public`.
+* The `Pragma: no-cache` header/value must not be present.
+* The `Set-Cookie` header must not be present.
+* `Vary` header parameters must be valid and not equal to `*`.
+
+>[!NOTE]
+> The Antiforgery system for generating secure tokens to prevent Cross-Site Request Forgery (CSRF) attacks will set the `Cache-Control` and `Pragma` headers to `no-cache` so that responses will not be cached.
+
+## HTTP response caching headers
+Response caching by the middleware is configured via your HTTP response headers. The relevant headers are listed below with notes on how they affect caching.
+
+Header | Details
+--- | --- |
+Authorization | <p>The response is not cached if the header exists.</p>
+Cache-Control | <p>The header must be present and its value must explicitly be marked `public`.</p><p>You can control caching with the following parameters:</p><ul><li>max-age</li><li>max-stale</li><li>min-fresh</li><li>must-revalidate</li><li>no-cache</li><li>no-store</li><li>only-if-cached</li><li>private</li><li>public</li><li>s-maxage</li><li>proxy-revalidate</li></ul>
+Pragma | <p>The value must not be `no-cache`.</p><p>Considered for backward compatibility with HTTP/1.0.</p>
```

Comment:
Should add the caveat that this header is overridden by the relevant headers in Cache-Control.